### PR TITLE
Hs119 update choose role logic

### PR DESCRIPTION
**Pull request by dlvinh**

### CODE REVIEW

File: lib/features/sign_up/view/choosing_roles_view.dart [hunk 1]
```diff
@@ -1,100 +1,118 @@
...
+                  Expanded(child: Container()),
+                  PrimaryButton(
+                    label: HatSpaceStrings.of(context).continueBtn,
+                    onPressed: state is UserRoleSelectedListState
+                        ? state.listRole.isNotEmpty
+                            ? () {
+                                _submitRoles();
+                              }
+                            : null
+                        : null,
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/choosing_roles_view.dart:103]: remove logic on UI

File: lib/features/sign_up/view/choosing_roles_view.dart [hunk 2]
```diff
@@ -1,100 +1,118 @@
...
+      create: (context) => ChooseRoleViewBloc(),
+      child: ChoosingRoleViewBody(),
+    );
+  }
 }
 
-class _ChoosingRolesState extends State<ChoosingRolesView> {
-  int totalNumberOfCheckedTextBoxes = 0;
+class ChoosingRoleViewBody extends StatelessWidget {
+  final List<Roles> listRole = [];
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/choosing_roles_view.dart:26]: if unused, please remove

File: lib/features/sign_up/view/choosing_roles_view.dart [hunk 3]
```diff
@@ -1,100 +1,118 @@
...
+    return BlocConsumer<ChooseRoleViewBloc, ChooseRoleViewState>(
+        listener: (context, state) {
+      // TODO: listen Success State
+    }, builder: (context, state) {
+      return Scaffold(
+          appBar: AppBar(
+            elevation: 0,
+            backgroundColor: HSColor.background,
+            leading: IconButton(
+              icon: const Icon(Icons.arrow_back, color: Colors.black),
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/choosing_roles_view.dart:46]: use HS color

File: lib/features/sign_up/view_model/choose_role_view_state.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+class StartListenRoleChange extends ChooseRoleViewState {
+  const StartListenRoleChange();
+  @override
+  List<Object> get props => [];
+}
+
+class UserRoleSelectedListState extends ChooseRoleViewState {
+  final Set<Roles> listRole;
+  const UserRoleSelectedListState(this.listRole);
+  Set<Roles> get listRoleProps => listRole;
```
Comment (hellomentorplus) [inline lib/features/sign_up/view_model/choose_role_view_state.dart:22]: duplicated get param, u can call `listRole` directly

File: lib/features/sign_up/view_model/choose_role_view_event.dart [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+abstract class ChooseRoleViewEvent extends Equatable {
+  const ChooseRoleViewEvent();
+}
+
+class OnChangeUserRoleEvent extends ChooseRoleViewEvent {
+  final int position;
+  const OnChangeUserRoleEvent(this.position);
+  @override
+  List<Object> get props => [];
```
Comment (hellomentorplus) [inline lib/features/sign_up/view_model/choose_role_view_event.dart:11]: update props value uniquely for this event

**Review by hellomentorplus — DISMISSED** [review]

**Review by suesitran — APPROVED** [review]

**Pull request merged**